src/Common/tests/System/Net/Http/LoopbackServer.cs

```diff
@@ -73,19 +74,29 @@ public static Task CreateServerAsync(Func<Socket, Uri, Task> funcAsync, out IPEn
                 .Where(a => a.IsIPv6LinkLocal)
                 .FirstOrDefault();
 
-        public static Task ReadRequestAndSendResponseAsync(Socket server, string response = null, Options options = null)
+        public static Task<List<string>> ReadRequestAndSendResponseAsync(Socket server, string response = null, Options options = null)
         {
             return AcceptSocketAsync(server, (s, stream, reader, writer) => ReadWriteAcceptedAsync(s, reader, writer, response), options);
         }
 
-        public static async Task ReadWriteAcceptedAsync(Socket s, StreamReader reader, StreamWriter writer, string response = null)
+        public static async Task<List<string>> ReadWriteAcceptedAsync(Socket s, StreamReader reader, StreamWriter writer, string response = null)
         {
-            while (!string.IsNullOrEmpty(await reader.ReadLineAsync().ConfigureAwait(false))) ;
+            // Read request line and headers. Skip any request body.
+            var lines = new List<string>();
+            string line = await reader.ReadLineAsync().ConfigureAwait(false);
+            while (!string.IsNullOrEmpty(line))
+            {
+                lines.Add(line);
+                line = await reader.ReadLineAsync().ConfigureAwait(false);
+            }
+
```

Comment:
Nit: Could be changed to:
```C#
string line;
while (!string.IsNullOrEmpty(line = await reader.ReadLineAsync().ConfigureAwait(false))
{
    lines.Add(line);
}
```
to avoid the duplicated line, but what you have is ok, too.